File: session07LinkedList/Stack.java
import java.util.ArrayList;

public class Stack {
    private ArrayList<Integer> impl;

    public Stack() {
        impl = new ArrayList<>();
    }
    public void push(int v) {
        impl.add(v); //O(1)
    }
    public void pop() { //O(1)
        impl.removeEnd();
    }
    public int last() {
        return impl.get(impl.size()-1);
    }
    public boolean isEmpty() { //O(1)
        return impl.size() == 0;
    }    
}

File: old/session06_GrowArray/Stack.cc
class Stack {
private:
	GrowArray impl;
public:
	void push(int v) { //O(1)
		impl.addEnd(v); //O(1)
	}
	int pop() {
		return impl.removeEnd(); //O(1)
	}

	bool isEmpty() const {
		return impl.size() == 0
	}
};

class Queue {
private:
  GrowArray impl;
public:
  void enqueue(int v) { //O(n)
    impl.addStart(v);
	}
	int dequeue() { //O(1)
    return imple.removeStart();
	}
	bool isEmpty() const { //O(1)
		return impl.size() = 0;
	}
}
